src/reconcile/rec_write.c

```diff
@@ -1390,13 +1371,9 @@ __rec_child_deleted(WT_SESSION_IMPL *session,
 	 * read into this part of the name space again, the cache read function
 	 * instantiates an entirely new page.)
 	 */
-	if (ref->addr != NULL &&
-	    (page_del == NULL ||
-	    __wt_txn_visible_all(session, page_del->txnid))) {
-		WT_RET(__wt_ref_info(session, ref, &addr, &addr_size, NULL));
-		WT_RET(__rec_block_free(session, addr, addr_size));
-		__wt_ref_free_addr(session, ref);
-	}
+	if (page_del == NULL ||
+	    __wt_txn_visible_all(session, page_del->txnid))
+		WT_RET(__wt_ref_block_free(session, ref));
```

Comment:
@michaelcahill, I'd put the test for `ref->addr != NULL` back in: if you don't the comment above it no longer makes sense, and this code parallels the code immediately below, which tests `ref->addr` again. Else: we rewrite the comment to explain how `ref->addr` is cleared by `__wt_ref_block_free`?